test/UID2.Client.Test/EncryptionTestsV2.cs

```diff
@@ -1,192 +1,183 @@
 ﻿using System;
 using System.Linq;
+using uid2_client.test.builder;
 using UID2.Client.Utils;
 using Xunit;
+using static uid2_client.test.TestData;
 
 namespace UID2.Client.Test
 {
     public class EncryptionTestsV2
     {
-        private static readonly long MASTER_KEY_ID = 164;
-        private static readonly long SITE_KEY_ID = 165;
-        private static readonly int SITE_ID = 9000;
-        private static readonly byte[] MASTER_SECRET = { 139, 37, 241, 173, 18, 92, 36, 232, 165, 168, 23, 18, 38, 195, 123, 92, 160, 136, 185, 40, 91, 173, 165, 221, 168, 16, 169, 164, 38, 139, 8, 155 };
-        private static readonly byte[] SITE_SECRET = { 32, 251, 7, 194, 132, 154, 250, 86, 202, 116, 104, 29, 131, 192, 139, 215, 48, 164, 11, 65, 226, 110, 167, 14, 108, 51, 254, 125, 65, 24, 23, 133 };
-        private static readonly DateTime NOW = DateTime.UtcNow;
-        private static readonly Key MASTER_KEY = new Key(MASTER_KEY_ID, -1, NOW.AddDays(-1), NOW, NOW.AddDays(1), MASTER_SECRET);
-        private static readonly Key SITE_KEY = new Key(SITE_KEY_ID, SITE_ID, NOW.AddDays(-10), NOW.AddDays(-1), NOW.AddDays(1), SITE_SECRET);
-        private static readonly string EXAMPLE_UID = "ywsvDNINiZOVSsfkHpLpSJzXzhr6Jx9Z/4Q0+lsEUvM=";
-        private static readonly string CLIENT_SECRET = "ioG3wKxAokmp+rERx6A4kM/13qhyolUXIu14WN16Spo=";
+        private const int SiteId = 12345;
+        private readonly UID2Client _client = new("endpoint", "authkey", CLIENT_SECRET, IdentityScope.UID2);
+        private readonly AdvertisingTokenBuilder _tokenBuilder = AdvertisingTokenBuilder.Builder().WithVersion(AdvertisingTokenBuilder.TokenVersion.V2);
```

Comment:
not sure if i misunderstand how unit testing works - so this stores a single AdvertisingTokenBuilder object. And in all the tests below it's modifying this object , NOT creating new AdvertisingTokenBuilder objects in/for each test right? So this object's lifetime lasts throughout every single unit test run, as this object isn't re-created for each unit test?

So if Unit Test A change the expiry of this AdvertisingTokenBuilder  object, wouldn't in the next test, say Unit Test B, be using the custom expiry set in Unit Test A? So there could be hidden dependency/change that test writer might not realise and hence testing the wrong assumptions?

Shouldn't we be safe and each test init its own custom AdvertisingTokenBuilder object?

Reply:
xUnit will create a new instance of the test class for each test

https://xunit.net/docs/shared-context#constructor

Reply:
ok crisis averted  lol

Reply:
Yep, sorry was in some meetings